src/main/java/com/depromeet/domain/feed/application/FeedService.java

```diff
@@ -0,0 +1,100 @@
+package com.depromeet.domain.feed.application;
+
+import com.depromeet.domain.feed.dto.response.FeedOneByProfileResponse;
+import com.depromeet.domain.feed.dto.response.FeedOneResponse;
+import com.depromeet.domain.follow.dao.MemberRelationRepository;
+import com.depromeet.domain.member.domain.Member;
+import com.depromeet.domain.mission.dao.MissionRepository;
+import com.depromeet.domain.mission.domain.Mission;
+import com.depromeet.domain.mission.domain.MissionVisibility;
+import com.depromeet.domain.missionRecord.dao.MissionRecordRepository;
+import com.depromeet.domain.missionRecord.domain.MissionRecord;
+import com.depromeet.global.util.MemberUtil;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+// TODO: Redis 사용해서 캐싱 작업 필요
+@Service
+@RequiredArgsConstructor
+public class FeedService {
+    private final MemberUtil memberUtil;
+    private final MissionRepository missionRepository;
+    private final MissionRecordRepository missionRecordRepository;
+    private final MemberRelationRepository memberRelationRepository;
+
+    @Transactional(readOnly = true)
+    public List<FeedOneResponse> findAllFeed() {
+        final Member currentMember = memberUtil.getCurrentMember();
+        List<Long> sourceIds =
+                new ArrayList<>(
+                        memberRelationRepository.findAllBySourceId(currentMember.getId()).stream()
+                                .map(memberRelation -> memberRelation.getTarget().getId())
+                                .toList());
+        sourceIds.add(currentMember.getId());
+
+        return missionRecordRepository.findFeedAll(sourceIds);
+    }
+
+    public List<FeedOneByProfileResponse> findAllFeedByTargetId(Long targetId) {
+        Member sourceMember = memberUtil.getCurrentMember();
+
+        /*
+         본인 프로필 findFeedBySourceMember
+         targetId가 본인이다.
+         그렇다면 공개 여부 상관없이 full-scan
+
+         타인 프로필 findFeedByTargetMember
+         만약 sourceId와 targetId가 다르다.
+         그렇다면 공개 여부에서 비공개는 exclude 한다.
+         추가로 내가 타인 프로필에서 팔로잉 관계가 이뤄졌는지 여부 확인하여 팔로우 관계 여부 체크
+        */
+        if (!targetId.equals(sourceMember.getId())) {
+            return findFeedByTargetMember(sourceMember, targetId);
+        } else {
+            return findFeedBySourceMember(sourceMember);
+        }
+    }
+
+    private List<FeedOneByProfileResponse> findFeedByTargetMember(
+            Member sourceMember, Long targetId) {
+        final Member targetMember = memberUtil.getMemberByMemberId(targetId);
+        List<MissionVisibility> visibilities = new ArrayList<>(List.of(MissionVisibility.ALL));
```

Comment:
List.of로 생성한 리스트는 Immutable해서 요렇게 쓰신건가용?

그렇다라고 한다면
```suggestion
        List<MissionVisibility> visibilities = new ArrayList<>();
        visiblities.add(MissionVisibility.ALL);
```
도 좋아보이네용